docs/tutorials/01_neural_networks.ipynb

```diff
@@ -121,339 +165,858 @@
+    "sampler_qnn_weights = algorithm_globals.random.random(sampler_qnn.num_weights)"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": 10,
-   "id": "moved-matrix",
+   "id": "a008cebc",
    "metadata": {},
    "outputs": [
     {
-     "data": {
-      "text/plain": [
-       "(None,\n",
-       " array([[[0.63272767]],\n",
-       " \n",
-       "        [[0.63272767]]]))"
-      ]
-     },
-     "execution_count": 10,
-     "metadata": {},
-     "output_type": "execute_result"
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Number of input features for SamplerQNN: 2 \n",
+      "Input: [0.85859792 0.69736803]\n",
+      "Number of trainable weights for SamplerQNN: 4 \n",
+      "Weights: [0.09417735 0.97562235 0.7611397  0.78606431]\n"
+     ]
     }
    ],
    "source": [
-    "# QNN batched backward pass\n",
-    "qnn1.backward([input1, input1], weights1)"
+    "print(\n",
+    "    f\"Number of input features for SamplerQNN: {sampler_qnn.num_inputs} \\nInput: {sampler_qnn_input}\"\n",
+    ")\n",
+    "print(\n",
+    "    f\"Number of trainable weights for SamplerQNN: {sampler_qnn.num_weights} \\nWeights: {sampler_qnn_weights}\"\n",
+    ")"
    ]
   },
   {
    "cell_type": "markdown",
-   "id": "completed-dressing",
+   "id": "7f1500df",
    "metadata": {},
    "source": [
-    "Combining multiple observables in a list allows to create more complex QNNs."
+    "Once we have the inputs and the weights, let's see the results for batched and non-batched passes."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "de7e7302",
+   "metadata": {},
+   "source": [
+    "### 3.2. Non-batched Forward Pass"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "52c5fcc5",
+   "metadata": {},
+   "source": [
+    "#### 3.2.1. `EstimatorQNN` Example"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7fba01a3",
+   "metadata": {},
+   "source": [
+    "For the `EstimatorQNN`, the expected output shape for the forward pass is `(1, num_qubits * num_observables)` where `1` in our case is the number of samples:"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": 11,
-   "id": "academic-roads",
+   "id": "54bed89e",
    "metadata": {},
-   "outputs": [],
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Forward pass result for EstimatorQNN: [[0.2970094]]. Shape : (1, 1)\n"
+     ]
+    }
+   ],
    "source": [
-    "observable2 = SparsePauliOp.from_list([(\"Z\" * qc1.num_qubits, 1)])\n",
+    "estimator_qnn_forward = estimator_qnn.forward(estimator_qnn_input, estimator_qnn_weights)\n",
     "\n",
-    "qnn2 = EstimatorQNN(\n",
-    "    circuit=qc1,\n",
-    "    observables=[observable1, observable2],\n",
-    "    input_params=[params1[0]],\n",
-    "    weight_params=[params1[1]],\n",
+    "print(\n",
+    "    f\"Forward pass result for EstimatorQNN: {estimator_qnn_forward}. Shape : {estimator_qnn_forward.shape}\"\n",
```

Comment:
Shapes get lost in the output. I don't know if a new line "\n" before printing the shape would help or add more confusion when a few lines are printed.

Reply:
Tried adding `\n`, I think it's readable enough.